mlflow/store/artifact/http_artifact_repo.py

```diff
@@ -119,3 +137,46 @@ def abort_multipart_upload(self, local_file, upload_id, artifact_path=None):
         }
         resp = http_request(host_creds, endpoint, "POST", json=params)
         augmented_raise_for_status(resp)
+
+    def _try_multipart_upload(self, local_file, artifact_path=None):
+        """
+        Attempts to perform multipart upload to log an artifact.
+        Returns if the multipart upload is successful.
+        Raises UnsupportedMultipartUploadException if multipart upload is unsupported.
+        """
+        parts = []
+        chunk_size = MLFLOW_MULTIPART_UPLOAD_CHUNK_SIZE.get()
+        size = os.path.getsize(local_file)
+        num_parts = math.ceil(size / chunk_size)
+
+        try:
+            create = self.create_multipart_upload(local_file, num_parts, artifact_path)
+        except HTTPError as e:
+            # return False if server does not support multipart upload
+            error_message = e.response.json().get("message", "")
+            if isinstance(error_message, str) and error_message.startswith(
+                "Multipart upload is not supported for artifact repository"
+            ):
+                raise UnsupportedMultipartUploadException(error_message)
+            else:
+                raise
+
+        try:
+            for i, credential in enumerate(create.credentials):
+                with open(local_file, "rb") as f:
+                    f.seek(i * chunk_size)
+                    chunk = f.read(chunk_size)
+
+                response = requests.put(credential.url, data=chunk)
+                augmented_raise_for_status(response)
+                parts.append(
+                    MultipartUploadPart(
+                        part_number=credential.part_number,
+                        etag=response.headers["ETag"],
+                    )
+                )
+
+            self.complete_multipart_upload(local_file, create.upload_id, parts, artifact_path)
+        except Exception:
+            self.abort_multipart_upload(local_file, create.upload_id, artifact_path)
```

Comment:
```suggestion
            _logger.warning("...")
            self.abort_multipart_upload(local_file, create.upload_id, artifact_path)
```

Can we add a warning message here?

Reply:
sure